pkg/payload/task_graph.go

```diff
@@ -536,7 +536,7 @@ func RunGraph(ctx context.Context, graph *TaskGraph, maxParallelism int, fn func
 	incompleteCount := 0
 	for i, result := range results {
 		if result == nil {
-			if firstIncompleteNode == nil {
+			if firstIncompleteNode == nil && len(graph.Nodes[i].Tasks) > 0 {
```

Comment:
if you do this, the check on line 548 needs to consider the case where firstIncompleteNode is  nil and the incompleteCount is greater than zero.

Reply:
[Line 548][1] already has a guard for `firstIncompleteNode != nil`.  If `incompleteCount` is greater than zero, and `firstIncompleteNode` is `nil`, I'm not sure we care, since no meaningful work was left on the table.  But I can check `nextNode` to make sure...

[1]: https://github.com/openshift/cluster-version-operator/blob/dc6f4f07d8986272b538b5d6bed9e8b6412b3ce6/pkg/payload/task_graph.go#L548

Reply:
I am still not sure why the Task i.e. `.Tasks[0]` is nil.

Reply:
but you don't produce the error in that case

Reply:
> I am still not sure why the Task i.e. `.Tasks[0]` is nil.

We have empty placeholder nodes in [the graph][1], like:

![Screenshot 2021-02-01 at 15 11 34](https://user-images.githubusercontent.com/209920/106531740-abd14e80-64a3-11eb-910f-c54681b13efe.png)

> But I can check `nextNode` to make sure...

I'm pretty sure we're fine with the change I'm proposing, without needing to involve `nextNode`.  Cases:

* All result-less nodes have no tasks.  Then with my change I will never set `firstIncompleteNode`, and we will not inject an `incomplete task nodes, beginning with...` error.  But there were no outstanding tasks, so not injecting that error is an improvement.
* The first result-less node has at least one task.  Then that matches my guard, and we get exactly the same non-panicky `firstIncompleteNode` handling as we have in master today.
* The first result-less node has no tasks, but some later result-less node has at least one task.  Then that later node will match my guard, and we'll mention it with a `incomplete task nodes, beginning with...` error instead of panicking (where we panic today).

[1]: https://github.com/openshift/cluster-version-operator/blob/1d30d9c07c4220d0271a37f5e8999316f67c3ce6/docs/user/reconciliation.md#manifest-graph